src/plans/create/steps/storage-map/utils.ts

```diff
@@ -0,0 +1,99 @@
+import type { InventoryStorage } from 'src/modules/Providers/hooks/useStorages';
+import { mapSourceStoragesToLabels } from 'src/modules/Providers/views/migrate/reducer/mapSourceToLabels';
+
+import type {
+  OpenstackVolume,
+  OVirtDisk,
+  ProviderVirtualMachine,
+  V1beta1Provider,
+} from '@kubev2v/types';
+
+import type { SourceProviderMappingLabels } from '../../types';
+
+import { StorageMapFieldId, type StorageMapping } from './constants';
+
+type StorageMappingId = `${StorageMapFieldId.StorageMap}.${number}.${keyof StorageMapping}`;
+
+export const getStorageMapFieldId = (id: keyof StorageMapping, index: number): StorageMappingId =>
+  `${StorageMapFieldId.StorageMap}.${index}.${id}`;
+
+const getStoragesUsedBySelectedVms = (
+  sourceStorageLabelToId: Record<string, string>,
+  selectedVMs: ProviderVirtualMachine[],
+  disks: (OVirtDisk | OpenstackVolume)[],
+): string[] =>
+  Array.from(
+    new Set(
+      selectedVMs.flatMap((vm) => {
+        switch (vm.providerType) {
+          case 'vsphere': {
+            return vm.disks?.map((disk) => disk?.datastore?.id);
+          }
+          case 'openstack': {
+            const vmDisks =
+              vm.attachedVolumes?.map((av) =>
+                (disks as OpenstackVolume[]).find((disk) => disk.id === av.ID),
+              ) ?? [];
+            const volumeTypeIds: string[] = vmDisks
+              ? vmDisks.map((disk) => disk?.volumeType ?? '')
+              : [];
+
+            if (vm?.imageID) {
+              volumeTypeIds.push('glance');
+            }
+
+            return volumeTypeIds;
+          }
+          case 'ovirt': {
+            const vmDisks = vm.diskAttachments?.map((da) =>
+              (disks as OVirtDisk[]).find((disk) => disk.id === da.disk),
+            );
+            const storageDomainIds: string[] = vmDisks
+              ? vmDisks.map((disk) => disk?.storageDomain ?? '')
+              : [];
+
+            return storageDomainIds;
+          }
+          case 'ova': {
+            return [Object.values(sourceStorageLabelToId)[0]];
+          }
+          case 'openshift':
+          case undefined:
+          default:
+            return [];
+        }
```

Comment:
Could you please split it and make it more readable?

Reply:
I split out those cases > 1 line to separate utils in this file.